### PR TITLE
Some updates of lottery module

### PR DESCRIPTION
**Pull request by tolak**
- Split lottery command into two events
- Move transfer_lottery to impl block and add sudo method for test
- Decode command before send to  TEE

### CODE REVIEW

File: pallets/bridge_transfer/src/lib.rs [hunk 1]
```diff
@@ -123,11 +110,62 @@ decl_module! {
...
+				));
+			}
 
 			Ok(())
 		}
 	}
 }
+
+impl<T: Config> Module<T> {
+	pub fn transfer_lottery(round_id: u32, token_id: u32, payload: Vec<u8>, dest_id: bridge::ChainId) -> DispatchResult {
```
Comment (jasl) [inline pallets/bridge_transfer/src/lib.rs:150]: Consider rename to `do_transfer_lottery`, follow the pattern https://github.com/paritytech/polkadot/blob/master/runtime/common/src/paras_registrar.rs#L219

Comment (tolak) [reply pallets/bridge_transfer/src/lib.rs]: Reamed

File: pallets/bridge_transfer/src/lib.rs [hunk 2]
```diff
@@ -48,8 +49,10 @@ decl_storage! {
 
 decl_event! {
 	pub enum Event {
-		/// Received BTC lottery encoded command
-		LotteryCommand(Vec<u8>),
+		/// Receive command: Newround. [roundId, totalCount, winnerCount]
+		LotteryNewround(u32, u32, u32),
```
Comment (jasl) [inline pallets/bridge_transfer/src/lib.rs:53]: LotteryNewRound

File: pallets/bridge_transfer/src/lib.rs [hunk 3]
```diff
@@ -48,8 +49,10 @@ decl_storage! {
...
+		/// Receive commnad: Openbox. [roundId, tokenId, btcAddress]
+		LotteryOpenbox(u32, u32, Vec<u8>),
```
Comment (jasl) [inline pallets/bridge_transfer/src/lib.rs:55]: LotteryOpenBox

File: pallets/bridge_transfer/src/lib.rs [hunk 4]
```diff
@@ -123,11 +110,62 @@ decl_module! {
...
+					metadata.len() == (13 + address_len),
+					Error::<T>::InvalidCommand
+				);
+
+				Self::deposit_event(Event::LotteryOpenbox(
+					u32::from_be_bytes(<[u8; 4]>::try_from(&metadata[1..5]).map_err(|_| Error::<T>::InvalidCommand)?),	// roundId
+					u32::from_be_bytes(<[u8; 4]>::try_from(&metadata[5..9]).map_err(|_| Error::<T>::InvalidCommand)?),	// tokenId
+					metadata[13..].to_vec()						// btcAddress
+				));
+			}
```
Comment (jasl) [inline pallets/bridge_transfer/src/lib.rs:142]: add `else` to cover other unexpected cases

**Review by h4x3rotab — APPROVED** [review]

**Comment by h4x3rotab** [comment]
LGTM

File: pallets/bridge_transfer/src/lib.rs [hunk 5]
```diff
@@ -123,11 +110,64 @@ decl_module! {
...
+					Error::<T>::InvalidCommand
+				);
+
+				Self::deposit_event(Event::LotteryOpenBox(
+					u32::from_be_bytes(<[u8; 4]>::try_from(&metadata[1..5]).map_err(|_| Error::<T>::InvalidCommand)?),	// roundId
+					u32::from_be_bytes(<[u8; 4]>::try_from(&metadata[5..9]).map_err(|_| Error::<T>::InvalidCommand)?),	// tokenId
+					metadata[13..].to_vec()						// btcAddress
+				));
+			} else {
+				// ignore
```
Comment (jasl) [inline pallets/bridge_transfer/src/lib.rs:143]: I suggest `fail!(Error::<T>::InvalidCommand)`

**Pull request merged**